add zap logger to controllers: pod, gwclass, gw, service

mikhail-aws opened the pull request:
Note:
- use zap logger vs wrapped version - zapr
- change "v" flag to "debug" to enable debug log
- use only 2 levels of logging info and debug
- replace glog with zap in following controllers: pod, service, gwclass, gateway
- partly address #126 

Logs example:
```
2023-08-09T20:56:50.126-0700	INFO	setup	aws-application-networking-k8s/main.go:97	init config	{"VpcId": "vpc-02aeb505678b5aad7", "Region": "us-west-2", "AccoundId": "566883718281", "DefaultServiceNetwork": "", "UseLongTgName": false}
...
2023-08-09T20:56:51.519-0700	DEBUG	datastore	latticestore/latticestore.go:169	AddServiceNetwork name:[my-hotel], account[566883718281] arn[arn:aws:vpc-lattice:us-west-2:566883718281:servicenetwork/sn-05bd7ca29c9532dcf], id[sn-05bd7ca29c9532dcf]
2023-08-09T20:56:51.519-0700	DEBUG	cloud	aws/cloud.go:26	request	{"serviceName": "VpcLattice", "operation": "ListServiceNetworks", "params": "{\n\n}"}
2023-08-09T20:56:51.589-0700	DEBUG	cloud	aws/cloud.go:38	response	{"serviceName": "VpcLattice", "operation": "ListServiceNetworks"}
2023-08-09T20:56:51.589-0700	INFO	controller.gateway	controllers/gateway_controller.go:208	successfully deployed model	{"stack": "my-hotel:default"}
2023-08-09T20:56:51.620-0700	DEBUG	controller.gateway	controllers/gateway_controller.go:238	serviceNetworkStatus : {{my-hotel 566883718281} arn:aws:vpc-lattice:us-west-2:566883718281:servicenetwork/sn-05bd7ca29c9532dcf sn-05bd7ca29c9532dcf service network is created and associated} for my-hotel
```

rlymbur commented:
Thank you for tackling this. As a precaution, do E2E tests continue to pass?

mikhail-aws commented:
> Thank you for tackling this. As a precaution, do E2E tests continue to pass?

I didnt run e2etests since only log statements are changed. But I ran a couple of manual testing steps.
```
make run
kubectl apply -f examples/my-hotel-gateway-tls.yaml
```

rlymbur commented on controllers/gateway_controller.go:
```diff
@@ -491,15 +474,13 @@ func UpdateGWListenerStatus(ctx context.Context, k8sclient client.Client, gw *ga
 	} else {
-		glog.V(2).Infof("no valid listeners for %v\n", gw.Name)
-		return errors.New("invalid listeners")
+		return fmt.Errorf("no valid listeners for %v\n", gw.Name)
```
nit: Is a new line still required here?

mikhail-aws replied:
no, missed \n, will fix other occurrences too

rlymbur commented on controllers/service_controller.go:
```diff
@@ -48,32 +47,39 @@ const (
 
-	latticeDataStore *latticestore.LatticeDataStore
-	stackMashaller   deploy.StackMarshaller
+	ds             *latticestore.LatticeDataStore
```
nit: Can we align these?

rlymbur commented on pkg/latticestore/latticestore.go:
```diff
@@ -166,13 +169,13 @@ func (ds *LatticeDataStore) AddServiceNetwork(name string, account string, arn s
 
 	if ok {
-		glog.V(6).Infof("UpdateServiceNetwork name:[%s], account[%s] arn[%s] id[%s]\n", name, account, arn, id)
+		ds.log.Debugf("UpdateServiceNetwork name:[%s], account[%s] arn[%s] id[%s]\n", name, account, arn, id)
```
nit: I notice these have new lines as well. Are these still required? I remember you mentioning these may not be anymore.

xWink commented on cmd/aws-application-networking-k8s/main.go:
```diff
@@ -72,27 +72,37 @@ func main() {
+		"AccoundId", config.AccountID,
+		"DefaultServiceNetwork", config.DefaultServiceNetwork,
+		"UseLongTgName", config.UseLongTGName,
+	)
```
Should we log every property individually here? My concern is that, as new init configs are added, this is one more place we need to keep updating. Can we just log all of the configs here (e.g. by logging the struct)?

mikhail-aws replied:
Config is package name here, and all properties are package global variables. There is a ticket in the backlog to make them into a struct, but there are many places that refer to these, it's a noticeable refactoring. #310

xWink commented on controllers/gateway_controller.go:
```diff
@@ -163,43 +160,43 @@ func (r *GatewayReconciler) reconcile(ctx context.Context, req ctrl.Request) err
 		return r.reconcileGatewayResources(ctx, gw)
 	} else {
-		gwLog.Info("Ignore non aws gateways!!!")
+		r.log.Infow("ignore non aws gateways!!!", "req", req)
```
Do we need the !!! in here?

mikhail-aws replied:
no idea!!!

xWink commented on controllers/service_controller.go:
```diff
@@ -48,32 +47,39 @@ const (
 	modelBuilder     gateway.LatticeTargetsBuilder
 	stackDeployer    deploy.StackDeployer
 
-	latticeDataStore *latticestore.LatticeDataStore
```
Why make this less readable?

mikhail-aws replied:
looks fine to me, but can return it back

xWink commented on pkg/utils/gwlog/gwlog_test.go:
```diff
@@ -0,0 +1 @@
+package gwlog
```
What's the purpose of this file?

mikhail-aws replied:
leftover after renaming log to gwlog, there was empty test file. will remove

xWink reviewed (commented):
Love this move towards better logging. Thank you!

solmonk reviewed (commented):
I know it is a too big of a PR to update and format every log lines, but at least let's make an announcement to get rid of `%v` and embed the structures whenever possible.

mikhail-aws commented on Makefile:
```diff
@@ -45,7 +45,7 @@ help: ## Display this help.
 .PHONY: run
 run: ## Run in development mode
-	go run cmd/aws-application-networking-k8s/main.go
+	go run cmd/aws-application-networking-k8s/main.go --debug
```
I'm thinking when you run controller in development mode you except it have debug logs.

mikhail-aws commented:
addressed comments and ran full e2e suite
```
Ran 11 of 11 Specs in 4965.560 seconds
SUCCESS! -- 11 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestIntegration (4966.49s)
```

rlymbur reviewed (approved):
Looks good to me!
